rate_limit_resources:
  - label: "connection_rate_limit"
    local:
      count: 100000
      interval: 1s

processor_resources:
  - label: "dimo_error_count"
    metric:
      type: counter
      name: dimo_error_count
      labels:
        component: ${! metadata("dimo_component") }
        source: ${! metadata("dimo_cloudevent_source") }
        error: ${! metadata("error") }

  - label: "dimo_partial_index_count"
    metric:
      type: counter
      name: dimo_partial_index_count
      labels:
        source: ${! metadata("dimo_cloudevent_source").or("unknown") }

  - label: "dimo_bad_request_sync_response"
    processors:
      - mapping: |
          meta response_status = 400
          root = metadata("response_message").or("Bad Request")
      - sync_response: {}

  - label: "dimo_internal_error_sync_response"
    processors:
      - mapping: |
          meta response_status = 500
          root  =  metadata("response_message").or("Internal Error: Please try again later")
      - sync_response: {}

  - label: "handle_db_error"
    processors:
      - group_by_value:
          value: ${! metadata("cloudevent_id").or("") } ${! metadata("cloudevent_source").or("") }
      - select_parts:
          parts: [0]
      - label: "log_db_error"
        log:
          level: WARN
          message: 'Dropping message'
          fields_mapping: |
            cloudevent_source = metadata("dimo_cloudevent_source").or("unknown")
            cloudevent_producer = metadata("dimo_cloudevent_producer").or("unknown")
            cloudevent_subject = metadata("dimo_cloudevent_subject").or("unknown")
            cloudevent_type = metadata("dimo_cloudevent_type").or("unknown")
            cloudevent_id = metadata("dimo_cloudevent_id").or("unknown")
            component = metadata("dimo_component").or("unknown")
            error = if errored() { error() } else { metadata("fallback_error").or("unknown") }
      - resource: "dimo_error_count"
      - resource: "dimo_internal_error_sync_response"
